src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountExtractPdpEntryServiceImpl.java

```diff
@@ -162,16 +165,104 @@ public Integer getPayeeNameFieldSize() {
     public String getConcurParameterValue(String parameterName) {
         String parameterValue = getParameterService().getParameterValueAsString(CUKFSConstants.ParameterNamespaces.CONCUR, 
                 CUKFSParameterKeyConstants.ALL_COMPONENTS, parameterName);
-        if (LOG.isDebugEnabled()) { 
-            LOG.debug("getConcurParamterValue, parameterName: " + parameterName + " paramterValue: " + parameterValue);
-        }
         return parameterValue;
     }
 
     public void setPayeeNameFieldSize(Integer payeeNameFieldSize) {
         this.payeeNameFieldSize = payeeNameFieldSize;
     }
     
+    @Override
+    public PdpFeedFileBaseEntry removeNonReimbursableSectionsFromPdpFeedFileBaseEntry(PdpFeedFileBaseEntry pdpFeedFileBaseEntry, 
+            ConcurStandardAccountingExtractBatchReportData reportData) {
+        LOG.debug("Entering removeNonReimbursableSectionsFromPdpFeedFileBaseEntry");
+        PdpFeedFileBaseEntry baseEntry = new PdpFeedFileBaseEntry();
+        baseEntry.setHeader(copyHeaderEntry(pdpFeedFileBaseEntry.getHeader()));
+        baseEntry.setVersion(pdpFeedFileBaseEntry.getVersion());
+        buildNewGroupEntries(pdpFeedFileBaseEntry, baseEntry);
+        baseEntry.setTrailer(buildPdpFeedTrailerEntry(baseEntry, reportData));
+        return baseEntry;
+    }
+
+    private void buildNewGroupEntries(PdpFeedFileBaseEntry pdpFeedFileBaseEntry, PdpFeedFileBaseEntry baseEntry) {
+        for (PdpFeedGroupEntry groupEntry : pdpFeedFileBaseEntry.getGroup()) {
+            PdpFeedGroupEntry newGroupEntry = copyGroupEntry(groupEntry);
+            boolean validNewGroup = false;
+            
+            for (PdpFeedDetailEntry detailEntry : groupEntry.getDetail()) {
+                PdpFeedDetailEntry newDetailEntry = copyDetailEntry(detailEntry);
+                KualiDecimal transactionTotal = KualiDecimal.ZERO;
+                for (PdpFeedAccountingEntry accountingEntry : detailEntry.getAccounting()) {
+                    transactionTotal = transactionTotal.add(accountingEntry.getAmount());
```

Comment:
Thank you for explaining this Jay, it makes sense now. 

Jay and I chatted about ways to improve this code so that this would be more clear. Jay is currently looking into re-working some things to add clarity.